roles/pool/src/lib/mod.rs

```diff
@@ -2,3 +2,105 @@ pub mod error;
 pub mod mining_pool;
 pub mod status;
 pub mod template_receiver;
+
+use async_channel::{bounded, unbounded};
+
+use mining_pool::{get_coinbase_output, Configuration, Pool};
+use template_receiver::TemplateRx;
+use tracing::{error, info, warn};
+
+use tokio::select;
+
+pub struct PoolSv2;
```

Comment:
on https://github.com/stratum-mining/stratum/pull/993, https://github.com/stratum-mining/stratum/pull/1095 and https://github.com/stratum-mining/stratum/pull/1092 we're introducing this pattern:

```
pub struct RoleName {
   config: Config
}
```

it would be good to keep that pattern consistent here